pkg/vsphere/session/session.go

```diff
@@ -155,14 +155,10 @@ func (s *Session) Connect(ctx context.Context) (*Session, error) {
 		}
 	}
 
+	soapURL.User = s.User
 	// Update the service URL with expanded defaults
 	s.Service = soapURL.String()
```

Comment:
Won't this  put the credentials in s.Service? I know you are nil-ing Service elsewhere, but this code is called from a  whole bunch of other places.